flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -869,41 +859,40 @@ public ApplicationReport startAppMaster(
 		FsPermission permission = new FsPermission(FsAction.ALL, FsAction.NONE, FsAction.NONE);
 		fs.setPermission(yarnFilesDir, permission); // set permission for path.
 
+		Path remoteYarnSiteXmlPath = null;
+		File f = new File(System.getenv("YARN_CONF_DIR"), Utils.YARN_SITE_FILE_NAME);
+		LOG.info("Adding Yarn configuration {} to the AM container local resource bucket", f.getAbsolutePath());
+		Path yarnSitePath = new Path(f.getAbsolutePath());
+		remoteYarnSiteXmlPath = setupSingleLocalResource(
+			Utils.YARN_SITE_FILE_NAME,
+			fs,
+			appId,
+			yarnSitePath,
+			localResources,
+			homeDir,
+			"");
+
 		//To support Yarn Secure Integration Test Scenario
-		//In Integration test setup, the Yarn containers created by YarnMiniCluster does not have the Yarn site XML
-		//and KRB5 configuration files. We are adding these files as container local resources for the container
+		//In Integration test setup, the Yarn containers created by YarnMiniCluster does not have the
+		//KRB5 configuration files. We are adding these files as container local resources for the container
 		//applications (JM/TMs) to have proper secure cluster setup
 		Path remoteKrb5Path = null;
-		Path remoteYarnSiteXmlPath = null;
 		boolean hasKrb5 = false;
-		if (System.getenv("IN_TESTS") != null) {
-			String krb5Config = System.getProperty("java.security.krb5.conf");
-			if (krb5Config != null && krb5Config.length() != 0) {
-				File krb5 = new File(krb5Config);
-				LOG.info("Adding KRB5 configuration {} to the AM container local resource bucket", krb5.getAbsolutePath());
-				Path krb5ConfPath = new Path(krb5.getAbsolutePath());
-				remoteKrb5Path = setupSingleLocalResource(
-					Utils.KRB5_FILE_NAME,
-					fs,
-					appId,
-					krb5ConfPath,
-					localResources,
-					homeDir,
-					"");
 
-				File f = new File(System.getenv("YARN_CONF_DIR"), Utils.YARN_SITE_FILE_NAME);
-				LOG.info("Adding Yarn configuration {} to the AM container local resource bucket", f.getAbsolutePath());
-				Path yarnSitePath = new Path(f.getAbsolutePath());
-				remoteYarnSiteXmlPath = setupSingleLocalResource(
-					Utils.YARN_SITE_FILE_NAME,
-					fs,
-					appId,
-					yarnSitePath,
-					localResources,
-					homeDir,
-					"");
-				hasKrb5 = true;
-			}
+		String krb5Config = System.getProperty("java.security.krb5.conf");
+		if (krb5Config != null && krb5Config.length() != 0) {
+			File krb5 = new File(krb5Config);
+			LOG.info("Adding KRB5 configuration {} to the AM container local resource bucket", krb5.getAbsolutePath());
+			Path krb5ConfPath = new Path(krb5.getAbsolutePath());
+			remoteKrb5Path = setupSingleLocalResource(
+				Utils.KRB5_FILE_NAME,
+				fs,
+				appId,
+				krb5ConfPath,
+				localResources,
+				homeDir,
+				"");
+			hasKrb5 = true;
```

Comment:
Shipping the Kerberos files via explicitly calling `addShipFiles` would be good as well. Then we could get rid of this code here.